### PR TITLE
Migrate to postgrator 3

### PR DESCRIPTION
**Pull request by dsvictor94**
Hello,
This week I investigated migration tools that uses only sql files and Postgrator was the one that best fit the requirements. However some features are not available in the current version of this client as I mentioned in issue #4 , so I took the liberty of migrating this client to Postgrator 3.

I tried to ensure that this change does not break projects that use the previous version, but I can not guarantee because I did not test on projects that already have migrations in version 2.

All tests passed, requiring minor modifications.

Thanks for making this tool available, I hope this pull request is useful.

### CODE REVIEW

File: test/postgrator-cli-tests.js [hunk 1]
```diff
@@ -219,10 +226,9 @@ var buildTestsForOptions = function (options) {
 
         postgratorCli.run(options, function(err, migrations) {
             restoreOptions();
-            deleteDefaultConfigFile();         
+            deleteDefaultConfigFile();
             assert.ifError(err);
-            assert.equal(migrations.length, 4);
-            assert.equal(migrations[migrations.length-1].version, 4);
+            assert.equal(migrations.length, 0);
```
Comment (MattiLehtinen) [inline test/postgrator-cli-tests.js]: Doesn't this change test assertion? Shouldn't it migrate 4 files here like before?

File: test/postgrator-cli-tests.js [hunk 2]
```diff
@@ -237,8 +243,7 @@ var buildTestsForOptions = function (options) {
         postgratorCli.run(options, function(err, migrations) {
             restoreOptions();
             assert.ifError(err);
-            assert.equal(migrations.length, 4);
-            assert.equal(migrations[migrations.length-1].version, 4);
+            assert.equal(migrations.length, 0);
```
Comment (MattiLehtinen) [inline test/postgrator-cli-tests.js]: Doesn't this change test assertion? Shouldn't it migrate 4 files here like before?

File: test/postgrator-cli-tests.js (hunk 1 above)
Comment (dsvictor94) [reply test/postgrator-cli-tests.js]: In version 3 of postgrator it only put the migrate object in the array when effectively execute the migration, if it only check the hash nothing is generated

File: test/postgrator-cli-tests.js (hunk 2 above)
Comment (dsvictor94) [reply test/postgrator-cli-tests.js]: In version 3 of postgrator it only put the migrate object in the array when effectively execute the migration, if it only check the hash nothing is generated

**Comment by MattiLehtinen** [comment]
Thanks for your contribution, appreciate it! And sorry about the long delay, I wish I have had more time for this.  

I made some slight modifications. E.g. when to show usage and tweaked tests to match more closely for the new postgrator functionality. Also bumped up major version to match postgrator's major version and to avoid breaking any projects. 

It is a major upgrade so I'll make some more testing before publishing it to npm.

**Pull request merged**